connector/src/main/java/org/geysermc/connector/network/translators/collision/CollisionManager.java

```diff
@@ -342,40 +322,37 @@ private Vector3d correctMovementForCollisions(Vector3d movement, BoundingBox bou
         BoundingBox movementBoundingBox = boundingBox.clone();
         movementBoundingBox.extend(movement);
 
-        List<Vector3i> collidableBlocks = getCollidableBlocks(movementBoundingBox);
-
         if (Math.abs(movementY) > CollisionManager.COLLISION_TOLERANCE) {
-            movementY = computeCollisionOffset(boundingBox, Axis.Y, movementY, collidableBlocks, checkWorld);
+            movementY = computeCollisionOffset(boundingBox, Axis.Y, movementY, movementBoundingBox, checkWorld);
             boundingBox.translate(0, movementY, 0);
         }
         boolean checkZFirst = Math.abs(movementZ) > Math.abs(movementX);
         if (checkZFirst && Math.abs(movementZ) > CollisionManager.COLLISION_TOLERANCE) {
-            movementZ = computeCollisionOffset(boundingBox, Axis.Z, movementZ, collidableBlocks, checkWorld);
+            movementZ = computeCollisionOffset(boundingBox, Axis.Z, movementZ, movementBoundingBox, checkWorld);
             boundingBox.translate(0, 0, movementZ);
         }
         if (Math.abs(movementX) > CollisionManager.COLLISION_TOLERANCE) {
-            movementX = computeCollisionOffset(boundingBox, Axis.X, movementX, collidableBlocks, checkWorld);
+            movementX = computeCollisionOffset(boundingBox, Axis.X, movementX, movementBoundingBox, checkWorld);
             boundingBox.translate(movementX, 0, 0);
         }
         if (!checkZFirst && Math.abs(movementZ) > CollisionManager.COLLISION_TOLERANCE) {
-            movementZ = computeCollisionOffset(boundingBox, Axis.Z, movementZ, collidableBlocks, checkWorld);
+            movementZ = computeCollisionOffset(boundingBox, Axis.Z, movementZ, movementBoundingBox, checkWorld);
             boundingBox.translate(0, 0, movementZ);
         }
 
         boundingBox.translate(-movementX, -movementY, -movementZ);
         return Vector3d.from(movementX, movementY, movementZ);
     }
 
-    private double computeCollisionOffset(BoundingBox boundingBox, Axis axis, double offset, List<Vector3i> collidableBlocks, boolean checkWorld) {
-        for (Vector3i blockPos : collidableBlocks) {
+    private double computeCollisionOffset(BoundingBox boundingBox, Axis axis, double offset, BoundingBox movementBoundingBox, boolean checkWorld) {
```

Comment:
Perhaps change this to the iterator and add a reset function to BlockPositionIterator so we can re-use the class?